### PR TITLE
Limit parallelism in ci-build

### PR DESCRIPTION
**Pull request by JunTaoLuo**
cc @pranavkm

### CODE REVIEW

File: makefile.shade [hunk 1]
```diff
@@ -121,7 +121,7 @@ var buildTarget = "compile"
             blockLogger.StartBlock("Cloning repos");
         }
 
-        Parallel.ForEach(repositories, repo =>
+        Parallel.ForEach(repositories, new ParallelOptions { MaxDegreeOfParallelism = threads }, repo =>
```
Comment (pranavkm) [inline makefile.shade]: Is `threads` defined in this target? Think you might have to copy the line from `ci-build`

Comment (JunTaoLuo) [inline makefile.shade]: Oops.

**Comment by pranavkm** [comment]
:shipit: 

Just run `build foo` to make sure the shade file actually compiles. (It should fail with key could not be found, and not a compilation dump)

**Comment by pranavkm** [comment]
Btw, could we get the fix into release. Ask @Eilon for ask mode approval

**Comment by muratg** [comment]
:shipit:

**Comment by JunTaoLuo** [comment]
Yup, `./build foo` gives `fail: The given key was not present in the dictionary.`. cc @Eilon for ask mode.

**Comment by JunTaoLuo** [comment]
Merged to release in https://github.com/aspnet/Universe/commit/be4d6bc07ccc5eaf9e1433463ea91f19a1843f69 and dev in https://github.com/aspnet/Universe/commit/55fe28286b0009a9b45f6070a76f7dcbd7bbefeb.

**Pull request closed without merging**